beacon-chain/blockchain/execution_engine.go

```diff
@@ -181,7 +181,7 @@ func (s *Service) optimisticCandidateBlock(ctx context.Context, blk block.Beacon
 	if err != nil {
 		return err
 	}
-	if parent == nil {
+	if parent == nil || parent.IsNil() {
```

Comment:
nogo was complaining here

Reply:
Use `helpers.BeaconBlockIsNil(parent)`?

Reply:
What was the complaint? Curious

Reply:
static analysis complaining of non-exhausting interface nilness check

Reply:
@terencechain there are a lot more of these instances I would rather fix up in a secondary, follow-up PR or some bigger PR that returns ErrNotFound instead of forcing nil checks